workflow/scripts/flow_generator.py

```diff
@@ -0,0 +1,100 @@
+
+from jinja2 import Environment, FileSystemLoader # templating engine
+import pandas as pd, os, argparse
+
+def get_parameters(data, name, crs=None):
+    if name != "bounding_box":
+        return ",".join(list(data[name].values.astype(str)))
+    elif name == "bounding_box" and crs is not None:
+        return get_bounding_box_list(data, crs)
+    
+def generate_bounding_box(row, crs):
+    if crs == "epsg3413":
+        columns = ["left_x", "top_y", "right_x", "lower_y"]
+    elif crs == "wgs84":
+        columns = ["top_left_lat", "top_left_lon", "lower_right_lat", "lower_right_lon"]
+    return "@".join(list(row[columns].values.astype(str)))
+
+def get_bounding_box_list(data, crs="wgs84"):
+    f = lambda x: generate_bounding_box(x, crs) 
+    return ",".join(data.apply(f, axis=1).values)
+
+def generate_cylc_file(csvfile="site_locations.csv", template="flow_template.j2", template_dir="./config", crs="wgs84", minfloearea=350, maxfloearea=90_000):
+    
+    df = pd.read_csv(csvfile)
+    date_columns = ['startdate', 'enddate']
+    df[date_columns] = df[date_columns].apply(lambda x: pd.to_datetime(x).dt.strftime('%Y-%m-%d'))
+    df['bounding_box'] = None
+    data = {c: get_parameters(df, c, crs) for c in df.columns}
+    data['rows'] = len(df.index)
```

Comment:
```suggestion
    data['rows'] = len(df.index) - 1
```
I think this is needed because the cycl task parameter starts at zero.

Reply:
Correct, good catch. I verified that the header row wasn't included, but forgot to check the end of file.